### PR TITLE
1.5.0 maa tasks

### PR DESCRIPTION
**Pull request by Gaben1025**
# MaA-Related Tasks
- [x] Adapt `culture_has_skirmisher_maa`, `culture_has_archer_maa`, etc.
- [x] Adapt new MaA types, `camel_cavalry` and `elephant_cavalry`
- [x] `KnightCulture` should depend on language instead of culture group
- [x] Use modifiers for mod MaA types in new modifiers
- [x] Adapt `GetRegionalInfantryMAA`, `GetRegionalInfantryMAA`, etc.
- [x] Adapt MaA types in `ep1_flavor.2000`
- [x] Finish Adapting all events 
image

### CODE REVIEW

**Comment by Gaben1025** [comment]
@zumbak04 What exactly is left to be adapted?
image

**Comment by Gaben1025** [comment]
@zumbak04 Is this also done?
image

image

**Comment by Gaben1025** [comment]
@zumbak04 Need elaboration on these 

image

**Comment by zumbak04** [comment]
> Use modifiers for mod MaA types in new modifiers

Most likely meant checking new modifiers (the ones in `common/modifiers`) and making sure they give bonuses to mod units along vanilla units).

> Adapt GetRegionalInfantryMAA, GetRegionalInfantryMAA, etc.

Should be pretty straigtforward. These are vanilla units and vanilla cultural parameters. Should be replaced by parameters from our cultural MaA traditions once we get them.

<details>
<summary>Click to expand</summary>

image

</details>

> Adapt MaA types in ep1_flavor.2000

Vanilla leftovers. Should be adapted to our cultures and tested.

<details>
<summary>Click to expand</summary>

image

</details>

File: events/dlc/ep1/ep1_flavor_events.txt [hunk 1]
```diff
@@ -2397,6 +2397,7 @@ scripted_effect ep1_flavor_2000_type_effect = {
 					OR = {
```
Comment (zumbak04) [inline events/dlc/ep1/ep1_flavor_events.txt:2397]: You missed a few parts in this file.
1. Set the flag, `flag:camel_cavalry`, etc. are set before. Our flags aren't.
image
2. A few usings of such flags in this file. Just seach for `flag:camel_cavalry` and do it by analogy.
image
3. `01_ep1_custom_loc.txt`, another file with `flag:camel_cavalry`. The same, do it by analogy.

File: localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ ep1_flavor.2000.desc_infantry:0 ""Our [ROOT.Char.Custom('GetRegionalInfantryMAA')|l] are vital to victory, they cannot be allowed to go soft.""
+ ep1_flavor.2000.desc_cavalry:0 ""Your [ROOT.Char.Custom('GetRegionalCavalryMAA')|l] are your most valuable asset, do not let them become sloppy.""
+ ep1_flavor.2000.desc_archer_cavalry:0 ""The mounted archers are our backbone; mounted bow competitions will keep them formidable.""
+ ep1_flavor.2000.desc_skirmishers:0 ""Our talented [ROOT.Char.Custom('GetRegionalSkirmisherMAA')|l] must compete, so that they always find their mark.""
+ ep1_flavor.2000.desc_elephant_cavalry:0 ""These fearsome creatures win us victory; they must be ready at all times.""
+ ep1_flavor.2000.desc_camel_cavalry:0 ""The vital camel riders are the core of your force, you must focus on them.""
+ ep1_flavor.2000.cavalry_tzykanion:0 ""The riders will play mounted team games!""
+ ep1_flavor.2000.cavalry_chovgan:0 ""Let them play mounted team games!""
+ ep1_flavor.2000.archer_cavalry:0 ""The riders will train with running rabbits.""
```
Comment (zumbak04) [inline localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml:15]: Why they are copied? I don't see any changes.

**Review by zumbak04 — CHANGES_REQUESTED** [review]

File: events/dlc/ep1/ep1_flavor_events.txt [hunk 3]
```diff
@@ -2747,6 +2831,29 @@ ep1_flavor.2000 = {
 			trigger = { scope:maa_type_1 = flag:levies }
 			text = ep1_flavor.2000.levies
 		}
+
```
Comment (zumbak04) [inline events/dlc/ep1/ep1_flavor_events.txt:2834]: `ep1_flavor.2000`, in `immediate` block there.s a section that analyzes your army: which MaAs you have.
Check every use of `number_maa_regiments_of_base_type` in this file. There should be mod MaA checks nearby.

**Review by zumbak04 — APPROVED** [review]
Hit approve, but pointed out one issue that needs to be addressed.

File: localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ # Martial Exercise
+ ep1_flavor.2000.desc_archer_cavalry:0 ""The mounted archers are our backbone; mounted bow competitions will keep them formidable.""
+ ep1_flavor.2000.desc_skirmishers:0 ""Our talented [ROOT.Char.Custom('GetRegionalSkirmisherMAA')|l] must compete, so that they always find their mark.""
+ ep1_flavor.2000.desc_elephant_cavalry:0 ""These fearsome creatures win us victory; they must be ready at all times.""
+ ep1_flavor.2000.cavalry_tzykanion:0 ""The riders will play mounted team games!""
+ ep1_flavor.2000.cavalry_chovgan:0 ""Let them play mounted team games!""
+ ep1_flavor.2000.archer_cavalry:0 ""The riders will train with running rabbits.""
+ 
+ ep1_flavor.2000.caster:0 ""The casters will test their focus and intellect to spar with each other.""
+ ep1_flavor.2000.massive_infantry:0 ""Our stoich giants will train their strength and grit.""
```
Comment (arithon) [inline localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml:12]:
```suggestion
 ep1_flavor.2000.massive_infantry:0 ""Our stoic giants will train their strength and grit.""
```

File: localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ ep1_flavor.2000.light_aerial:0 ""They will hold a race in the sky!.""
+ ep1_flavor.2000.massive_aerial:0 ""Training will help maneuver the skies.""
+ 
+ ep1_flavor.2000.desc_caster:0 ""The casters are a versitile set of fighters, training them will keep our edge sharp as ever.""
```
Comment (arithon) [inline localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml:16]:
```suggestion
 ep1_flavor.2000.desc_caster:0 ""The casters are a versatile set of fighters, training them will keep our edge sharp as ever.""
```

File: localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+ ep1_flavor.2000.desc_massive_infantry:0 ""The muscle of our ground forces needs to be prepared for whatever may come.""
+ ep1_flavor.2000.desc_light_aerial:0 ""Control of every aspect of the battlefield is important, these riders will help is mantain control of the skies.""
```
Comment (arithon) [inline localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml:18]:
```suggestion
 ep1_flavor.2000.desc_light_aerial:0 ""Control of every aspect of the battlefield is important, these riders will help us maintain control of the skies.""
```

File: localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+ ep1_flavor.2000.desc_massive_aerial:0 ""Death ffrom above must be guarenteed for our enemies; those providing should be evermore vigilant.""
```
Comment (arithon) [inline localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml:19]:
```suggestion
 ep1_flavor.2000.desc_massive_aerial:0 ""Death from above must be guaranteed for our enemies; those providing should be evermore vigilant.""
```

File: localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ 
+ # Interservice Rivalry
+ ep1_flavor.2020.desc_knight_1_caster:0 ""...is that all you got magician? Well for my next trick, I will make you #EMP disappear#!!""
```
Comment (arithon) [inline localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml:22]:
```suggestion
 ep1_flavor.2020.desc_knight_1_caster:0 ""...is that all you got, magician? Well for my next trick, I will make you #EMP disappear#!!""
```

File: localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+ ep1_flavor.2020.desc_knight_1_massive_infantry:0 ""...lots of brawn but no brains, huh? Either way, I will #EMP cut you down to size#!!"
+ ep1_flavor.2020.desc_knight_1_light_aerial:0 ""...might as well fly away, you have got #EMP no chance#!!"
+ ep1_flavor.2020.desc_knight_1_massive_aerial:0 ""...the bigger they are the harder they fall, especially from #EMP that height#!!"
```
Comment (arithon) [inline localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml:25]:
```suggestion
 ep1_flavor.2020.desc_knight_1_massive_aerial:0 ""...the bigger they are, the harder they fall, especially from #EMP that height#!!"
```

File: localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+ 
+ ep1_flavor.2020.desc_knight_2_caster:0 ""Ha! Thats rich coming from a rag wearing clown. Go drink your little potions!""
```
Comment (arithon) [inline localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml:27]:
```suggestion
 ep1_flavor.2020.desc_knight_2_caster:0 ""Ha! That's rich coming from a rag-wearing clown. Go drink your little potions!""
```

File: localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+ ep1_flavor.2020.desc_knight_2_massive_infantry:0 ""If you were normal size, you wouldn't be running that lip.""
+ ep1_flavor.2020.desc_knight_2_light_aerial:0 ""Pah! You got your head in the clouds, or up somewhere else if we were to be honest!""
```
Comment (arithon) [inline localization/english/dlc/ep1/wc_dlc_ep1_flavor_events_l_english.yml:29]:
```suggestion
 ep1_flavor.2020.desc_knight_2_light_aerial:0 ""Pah! You got your head in the clouds or up somewhere else, if we were to be honest!""
```

**Pull request merged**